LayoutTests/fast/scrolling/mac/smooth-scroll-recursive-frame-to-overflow.html

```diff
@@ -0,0 +1,71 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ useFlexibleViewport=true EventHandlerDrivenSmoothKeyboardScrollingEnabled=true ] -->
+
+<html>
+
+<head>
+    <script src="../../../resources/ui-helper.js"></script>
+    <script src="../../../resources/js-test-pre.js"></script>
+    <meta name="viewport" content="initial-scale=1.5, user-scalable=no">
+    <style>
+        body {
+            height: 2000px;
+        }
+        .scroller {
+            width: 20%;
+            height: 20%;
+            overflow: scroll;
+            border: 1px solid black;
+            padding: 10px;
+        }
+
+        #innerDiv {
+            height: 600px;
+        }
+
+    </style>
+    <script>
+        if (window.testRunner) {
+            testRunner.dumpAsText();
+            testRunner.waitUntilDone();
+        }
+
+        function documentScrolled()
+        {
+            debug("UNSUCCESSFUL: recursive scroll skipped over overflow:scroll region.");
+            testRunner.notifyDone();
+        }
+
+        function overflowScrolled()
+        {
+            debug("SUCCESSFUL: frame to overflow:scroll recursive scroll.");
```

Comment:
Same point about clarity in this wording. Generally "recursive" in my mind means "downward", but this is upward scroll propagation.